Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -67,6 +78,32 @@ Ref<FragmentedSharedBuffer> FragmentedSharedBuffer::create(DataSegment::Provider
     return adoptRef(*new FragmentedSharedBuffer(WTFMove(provider)));
 }
 
+#if PLATFORM(COCOA)
+std::optional<Ref<FragmentedSharedBuffer>> FragmentedSharedBuffer::fromIPCData(IPCData&& ipcData)
+{
+    return WTF::switchOn(WTFMove(ipcData), [](Vector<std::span<const uint8_t>>&& data) -> std::optional<Ref<FragmentedSharedBuffer>> {
+        size_t size = sizeof(data[0]) * data.size();
+        if (useUnixDomainSockets() || size < minimumPageSize) {
+            Ref sharedMemoryBuffer = SharedBuffer::create();
+            for (auto span : data)
+                sharedMemoryBuffer->append(span);
+            return sharedMemoryBuffer;
+        }
+        return std::nullopt;
+    }, [](std::optional<WebCore::SharedMemoryHandle>&& handle) -> std::optional<Ref<FragmentedSharedBuffer>> {
+        if (useUnixDomainSockets() || !handle.has_value() || (handle->size() && handle->size() < minimumPageSize))
```

Comment:
If the encoder returned an empty Vector in the empty case, we wouldn't need `handle->size() && `